pkg/injector/patch.go

```diff
@@ -19,12 +20,16 @@ import (
 func (wh *mutatingWebhook) createPatch(pod *corev1.Pod, req *v1beta1.AdmissionRequest, proxyUUID uuid.UUID) ([]byte, error) {
 	namespace := req.Namespace
 
+	// Tracks the success of the current certificate issued for envoy to connect to XDS
+	success := true
 	// Issue a certificate for the proxy sidecar - used for Envoy to connect to XDS (not Envoy-to-Envoy connections)
 	cn := catalog.NewCertCommonNameWithProxyID(proxyUUID, pod.Spec.ServiceAccountName, namespace)
+	defer certXdsIssueTimeTrack(time.Now(), cn.String(), &success)
```

Comment:
Isn't this going to lead to incorrect time tracking because of the defer statement? The time taken to issue the cert is not the same as the time taken for this function to complete. 

The webhook time tracking is done a bit differently because it calculates the entire time for the entrypoint function to execute, but in this case we should be calculating the time immediately before and after `IssueCertificate()` is invoked.

Reply:
That's right, I missed updating the logic. My latest commit has the necessary changes